packages/block-library/src/query-loop/index.php

```diff
@@ -46,27 +43,34 @@ function render_block_core_query_loop( $attributes, $content, $block ) {
 		}
 	}
 
-	$wrapper_attributes = get_block_wrapper_attributes( array( 'class' => $classnames ) );
+	if ( $query->have_posts() ) {
+		$wrapper_attributes = get_block_wrapper_attributes( array( 'class' => $classnames ) );
 
-	$content = '';
-	foreach ( $posts as $post ) {
-		$block_content = (
-			new WP_Block(
-				$block->parsed_block,
-				array(
-					'postType' => $post->post_type,
-					'postId'   => $post->ID,
+		$content = '';
+		while ( $query->have_posts() ) :
+			$query->the_post();
+			$block_content = (
+				new WP_Block(
+					$block->parsed_block,
+					array(
+						'postType' => get_post_type(),
+						'postId'   => get_the_ID(),
+					)
 				)
-			)
-		)->render( array( 'dynamic' => false ) );
-		$content      .= "<li>{$block_content}</li>";
+			)->render( array( 'dynamic' => false ) );
+			$content      .= "<li>{$block_content}</li>";
+		endwhile;
+
+		return sprintf(
+			'<ul %1$s>%2$s</ul>',
+			$wrapper_attributes,
+			$content
+		);
 	}
 
-	return sprintf(
-		'<ul %1$s>%2$s</ul>',
-		$wrapper_attributes,
-		$content
-	);
+	wp_reset_postdata();
```

Comment:
Shouldn't this be called if we have `$query->have_posts()` and after the end of the loop? That means loop the posts which call `$query->the_post();`, `wp_reset_postdata` and then return the content?

Reply:
yeah, that was an error on my part. fixed 👍